C/map.c

```diff
@@ -0,0 +1,251 @@
+#include <stdio.h>
+#include <malloc.h>
+#include <time.h>
+
+#include "loadingBar.h"
+#include "gradientGrid.h"
+#include "layer.h"
+#include "chunk.h"
+#include "map.h"
+
+double* getMapValue(map* map, int width_idx, int height_idx)
+{
+    double* map_value = NULL;
+
+    if (map != NULL)
+    {
+        int width = map->map_width * map->chunk_width;
+        int height = map->map_height * map->chunk_height;
+
+        if (width_idx < 0 || width_idx >= width)
+        {
+            printf("%sERROR : invalid width_idx = %d when reading map value. Should be in range [0, %d]%s\n", RED_COLOR, width_idx, width - 1, DEFAULT_COLOR);
+            return NULL;
+        }
+        
+        if (height_idx < 0 || height_idx >= height)
+        {
+            printf("%sERROR : invalid height_idx = %d when reading map value. Should be in range [0, %d]%s\n", RED_COLOR, height_idx, height - 1, DEFAULT_COLOR);
+            return NULL;
+        }
+
+        map_value = (map->map_values) + height_idx * width + width_idx;
+    }
+
+    return map_value;
+}
+
+
+
+chunk* getChunk(map* map, int width_idx, int height_idx)
+{
+    int width = map->map_width;
+    int height = map->map_height;
+
+    if (width_idx < 0 || width_idx >= width)
+    {
+        return NULL;
+    }
+    
+    if (height_idx < 0 || height_idx >= height)
+    {
+        return NULL;
+    }
+
+    return map->chunks[height_idx * width + width_idx];
+}
+
+
+
+
+
+map* newMapFromChunks(int map_width, int map_height, chunk* chunks[map_width * map_height], int display_loading)
+{
+    clock_t start_time = clock();
+
+    if (chunks != NULL && map_width > 0 && map_height > 0)
+    {
+        map* new_map = calloc(1, sizeof(map));
+
+        new_map->map_width = map_width;
+        new_map->map_height = map_height;
+
+        // copy chunks list to ensure dynamic allocation
```

Comment:
I am not sure I follow why you need to do so, and why a simple list copy rather than a 'deep copy'

Reply:
Nevermind I understood